internal/experiments/experiments_remote_execute_api.go

```diff
@@ -0,0 +1,201 @@
+}
+func (p *RemoteExecuteAPIExperimentConfig) Technique() string {
+	return categories.MITRE.Execution.NewContainer.Technique
+}
+func (p *RemoteExecuteAPIExperimentConfig) Tactic() string {
+	return categories.MITRE.Execution.NewContainer.Tactic
+}
+func (p *RemoteExecuteAPIExperimentConfig) Framework() string {
+	return string(categories.Mitre)
+}
+
+func (p *RemoteExecuteAPIExperimentConfig) Run(ctx context.Context, client *k8s.Client, experimentConfig *ExperimentConfig) error {
+	var config RemoteExecuteAPIExperimentConfig
+	yamlObj, _ := yaml.Marshal(experimentConfig)
+	err := yaml.Unmarshal(yamlObj, &config)
+	if err != nil {
+		return err
+	}
+
+	executorConfig := executor.NewExecutorConfig(
+		config.Metadata.Name,
+		config.Metadata.Namespace,
+		config.Parameters.Image,
+		config.Parameters.ServiceAccountName,
+		config.Parameters.TargetPort,
+		config.Parameters.LocalPort,
+	)
+
+	err = executorConfig.Deploy(ctx, client.Clientset)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *RemoteExecuteAPIExperimentConfig) Verify(ctx context.Context, client *k8s.Client, experimentConfig *ExperimentConfig) (*verifier.Outcome, error) {
+	var config RemoteExecuteAPIExperimentConfig
+	yamlObj, _ := yaml.Marshal(experimentConfig)
+	err := yaml.Unmarshal(yamlObj, &config)
+	if err != nil {
+		return nil, err
+	}
+
+	executorConfig := executor.NewExecutorConfig(
+		config.Metadata.Name,
+		config.Metadata.Namespace,
+		config.Parameters.Image,
+		config.Parameters.ServiceAccountName,
+		config.Parameters.TargetPort,
+		config.Parameters.LocalPort,
+	)
+
+	stopChan := make(chan struct{}, 1)
+	readyChan := make(chan struct{}, 1)
+	errch := make(chan error, 1)
+	out, errOut := new(bytes.Buffer), new(bytes.Buffer)
+
+	forwardConfig := executor.PortForwardConfig{
+		StopCh:  stopChan,
+		ReadyCh: readyChan,
+		Out:     out,
+		ErrOut:  errOut,
+	}
+
+	go func() {
+		err := executorConfig.OpenLocalPort(ctx, client, config.Parameters.LocalPort, forwardConfig)
+		if err != nil {
+			errch <- fmt.Errorf("Local Port Failed to open: %w", err)
+		}
+	}()
+
+	// Waits until local port is ready or open erros
+	select {
+	case <-readyChan:
+		break
+	case err := <-errch:
+		return nil, err
+	}
+
+	path := config.Parameters.Path
+	port := config.Parameters.LocalPort
+	url := url.URL{
+		Scheme: "http",
+		Host:   fmt.Sprintf("%s:%d", "127.0.0.1", port),
+		Path:   path,
+	}
+
+	v := verifier.New(
+		config.Metadata.Name,
+		config.Description(),
+		config.Framework(),
+		config.Tactic(),
+		config.Technique(),
+	)
+
+	body, statusCode, err := retrieveAPIResponse(url.String())
+	if err != nil {
+		return nil, err
+	}
+
+	if statusCode == config.Parameters.ExpectedHTTPResponse {
+		v.Success("ExpectedStatusCodeRecieved")
+	} else {
+		v.Fail("ExpectedStatusCodeRecieved")
+	}
+
+	if body == config.Parameters.ExpectedBody {
+		v.Success("ExpectedBodyRecieved")
```

Comment:
if we pass in the URLs in the config, maybe the results could be per URL/ hostname?